tests/robustness/checker/checker.go

```diff
@@ -276,43 +279,39 @@ func (chk *Checker) safeRestorePrepare(snapID string) (*SnapshotMetadata, error)
 // RestoreVerifySnapshot restores a snapshot and verifies its integrity against
 // the metadata provided.
 func (chk *Checker) RestoreVerifySnapshot(ctx context.Context, snapID, destPath string, ssMeta *SnapshotMetadata, reportOut io.Writer, opts map[string]string) error {
-	err := chk.snapshotIssuer.RestoreSnapshot(snapID, destPath, opts)
-	if err != nil {
-		return err
+	if ssMeta != nil {
+		return chk.snapshotIssuer.RestoreSnapshotCompare(snapID, destPath, ssMeta.ValidationData, reportOut, opts)
 	}
 
-	if ssMeta == nil && chk.RecoveryMode {
-		var b []byte
-
-		b, err = chk.validator.Gather(ctx, destPath, opts)
-		if err != nil {
-			return err
-		}
-
-		ssMeta = &SnapshotMetadata{
-			SnapID:         snapID,
-			ValidationData: b,
-		}
-
-		chk.mu.Lock()
-		defer chk.mu.Unlock()
-
-		err = chk.saveSnapshotMetadata(ssMeta)
-		if err != nil {
-			return err
-		}
+	// We have no metadata for this snapshot ID.
+	if !chk.RecoveryMode {
+		return robustness.ErrMetadataMissing
+	}
 
-		chk.SnapIDIndex.AddToIndex(snapID, allSnapshotsIdxName)
-		chk.SnapIDIndex.AddToIndex(snapID, liveSnapshotsIdxName)
+	// Recovery path:
+	// If in recovery mode, restore the snapshot by snapshot ID and gather
+	// its fingerprint (i.e. assume the snapshot will restore properly).
+	fingerprint, err := chk.snapshotIssuer.RestoreSnapshot(snapID, destPath, opts)
+	if err != nil {
+		return err
+	}
 
-		return nil
+	ssMeta = &SnapshotMetadata{
+		SnapID:         snapID,
+		ValidationData: fingerprint,
 	}
 
-	err = chk.validator.Compare(ctx, destPath, ssMeta.ValidationData, reportOut, opts)
+	chk.mu.Lock()
+	defer chk.mu.Unlock()
+
+	err = chk.saveSnapshotMetadata(ssMeta)
 	if err != nil {
 		return err
 	}
 
+	chk.SnapIDIndex.AddToIndex(snapID, allSnapshotsIdxName)
+	chk.SnapIDIndex.AddToIndex(snapID, liveSnapshotsIdxName)
+
 	return nil
 }
 
```

Comment:
It would be good if the 3 index names (`deletedSnapshotsIdxName`, `liveSnapshotsIdxName`,  `allSnapshotsIdxName`)  be exposed (capitalized) so that the repo can be introspected without changing code in this module.

Reply:
Done! 👍